File: fix_syntax.js
const fs = require('fs');
let code = fs.readFileSync('src/store/cadStore.ts', 'utf8');

// I need to change addEntity and importDxfData to have `{ set(`
code = code.replace(
  /addEntity: \(sketchIdOrEntity: any, entity\?: any\) => set\(\(state\) => \{/g,
  `addEntity: (sketchIdOrEntity: any, entity?: any) => { set((state) => {`
);

code = code.replace(
  /importDxfData: \(entities, layers\) => set\(\(state\) => \{/g,
  `importDxfData: (entities, layers) => { set((state) => {`
);

fs.writeFileSync('src/store/cadStore.ts', code);

File: fix_syntax.py
with open('src/store/cadStore.ts', 'r') as f:
    code = f.read()

code = code.replace(
    '  toggleShow3DEdges: () => {\n    set((state) => { state.show3DEdges = !state.show3DEdges; }),',
    '  toggleShow3DEdges: () => set((state) => { state.show3DEdges = !state.show3DEdges; }),'
)

code = code.replace(
"""
    });

    },
  removeFeature: (id) => set((state) => {
""",
"""
    });
  },
  removeFeature: (id) => set((state) => {
"""
)

with open('src/store/cadStore.ts', 'w') as f:
    f.write(code)

